Xcode 11.3 is gone from macOS-latest (=macOS-11)

## Description
<!--
If your issue is a bugreport, this means describing what you did,
what did you want to happen and what actually did happen.

If your issue is a feature request, describe the feature and why do you
want it.
-->

### Steps to reproduce
<!--
This is only relevant for bug reports, but if you do have one,
please provide a minimal set of steps to reproduce the problem.

Usually this means providing a small and self-contained code using doctest
and specifying compiler flags/tools used if relevant.
-->
1. Create a PR or commit that triggers a build

#### Results
https://github.com/onqtam/doctest/runs/4114874846?check_suite_focus=true

> xcode-select: error: invalid developer directory '/Applications/Xcode_11.3.app'
Error: Process completed with exit code 1.

### Extra information
<!--
Fill in any extra information that might be important for your issue.

If your issue is a bugreport, definitely fill out at least the following.
-->
* doctest version: **dev** (7810b9e3d2faedb2dcf81834da2be8d3025c022a)
* Operating System: **macOS 11.6.1**
* Compiler+version: **XCode 11.3** (sorta, it's requested by CI, but not available)

https://github.com/actions/virtual-environments/issues/4060

--- a/.github/workflows/main.yml
+++ b/.github/workflows/main.yml
@@ -194,7 +194,7 @@ jobs:
             compiler: gcc
 
           - name: macOS-latest-xcode-11.3
-            os: macOS-latest
+            os: macOS-10.15
             compiler: xcode
             version: "11.3"
 

